src/app/content/highlights/components/SummaryPopup/FiltersList.tsx

```diff
@@ -36,69 +36,59 @@ interface FiltersListColorProps {
 }
 
 // tslint:disable-next-line: variable-name
-export const FiltersListColor = (props: FiltersListColorProps) => {
-  const handleClick = () => {
-    props.onRemove(props.color);
-  };
-
-  return <li>
-    <RemoveIcon onClick={handleClick}><Times /></RemoveIcon>
+export const FiltersListColor = (props: FiltersListColorProps) => (
+  <li>
+    <StyledPlainButton onClick={() => props.onRemove(props.color)}><Times /></StyledPlainButton>
     <ItemLabel>
       <FormattedMessage id={`i18n:highlighting:colors:${props.color}`}>
         {(msg: Element | string) => msg}
       </FormattedMessage>
     </ItemLabel>
-  </li>;
-};
+  </li>
+);
 
 interface FiltersListChapterProps {
   title: string;
-  chapterId: string;
+  locationId: string;
   onRemove: (chapterId: string) => void;
 }
 
 // tslint:disable-next-line: variable-name
-export const FiltersListChapter = (props: FiltersListChapterProps) => {
-  const handleClick = () => {
-    props.onRemove(props.chapterId);
-  };
-
-  return <li>
-    <RemoveIcon onClick={handleClick}><Times /></RemoveIcon>
+export const FiltersListChapter = (props: FiltersListChapterProps) => (
+  <li>
+    <StyledPlainButton onClick={() => props.onRemove(props.locationId)}><Times /></StyledPlainButton>
     <ItemLabel dangerouslySetInnerHTML={{ __html: props.title }} />
-  </li>;
-};
+  </li>
+);
 
 interface FiltersListProps {
   className?: string;
 }
 
 // tslint:disable-next-line: variable-name
 const FiltersList = ({className}: FiltersListProps) => {
-  const locations = useSelector(highlightLocations);
+  const locationFilters = useSelector(highlightLocationFilters);
   const filters = useSelector(summaryFilters);
 
   const dispatch = useDispatch();
 
   const onRemoveChapter = (locationId: string) => {
     dispatch(setSummaryFilters({
-      ...filters,
       locationIds: filters.locationIds.filter(not(match(locationId))),
     }));
   };
 
   const onRemoveColor = (color: HighlightColorEnum) => {
     dispatch(setSummaryFilters({
-      ...filters,
       colors: filters.colors.filter(not(match(color))),
     }));
   };
 
   return <ul className={className}>
-    {filters.locationIds.map((locationId) => locations.has(locationId) && <FiltersListChapter
+    {filters.locationIds.map((locationId) => locationFilters.has(locationId) && <FiltersListChapter
       key={locationId}
-      title={locations.get(locationId)!.title}
-      chapterId={locationId}
+      title={locationFilters.get(locationId)!.title}
+      locationId={locationId}
       onRemove={onRemoveChapter}
```

Comment:
if you make this `onRemove={onRemoveChapter(locationId)}` and change the handler to 

```
const onRemoveChapter = (locationId: string) => () => {
    dispatch(setSummaryFilters({
      ...filters,
      locationIds: filters.locationIds.filter(not(match(locationId))),
    }));
  };
```

then the `FiltersListChapter` component doesn't need any knowledge of the `locationId`

Reply:
i'm not sure if i haven't been explaining myself well or if you're just ignoring me because you really really hate currying, but regardless its fine, i think everything is looking good now.

Reply:
I'm confused, I think I've applied your suggestion with one change:
I did `onRemove={() => onRemoveChapter(locationId)}` instead of `onRemove={onRemoveChapter(locationId)}` because with the second option `onRemoveChapter` function will be called immediately after render

Reply:
not if you also change the definition as i put in the comment so that the initial call returns a new function that takes no arguments, the second function is then invoked by the handler. this way of stashing data in the parameters of previously called functions is called currying, although this isn't a perfect example.

Reply:
Ahh, I wasn't looking at `=> () =>`